src/main/java/org/testeditor/fixture/web/json/BrowserSetupReader.java

```diff
@@ -0,0 +1,147 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2017 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+
+package org.testeditor.fixture.web.json;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.testeditor.fixture.web.io.FileReader;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+public class BrowserSetupReader {
+    
+    private static final String optionsString = "options";
+    private static final String capabilitiesString = "capabilities"; 
+
+    /**
+     * Reads all available elements. An element is a {@link BrowserSetupElement} like operating system "os" with the 
+     * value "WINDOWS" or "LINUX"s or browser specific "capabilities" or "options". These elements can be used for
+     * setting up a browser before starting a test.
+     * @param fileName of the BrowserElements
+     * @return All Browser or OS specific settings for starting a browser in a test environment.  
+     */
+    public ArrayList<BrowserSetupElement> readElements(String fileName) {
+        FileReader reader = new FileReader();
+        String jsonString = reader.getFileContentAsString(fileName);
+        Gson gson = new GsonBuilder().create();
+        JsonObject obj = gson.fromJson(jsonString, JsonObject.class);
+        ArrayList<BrowserSetupElement> allSetupElements = getAllSetupElements(obj);
+        extractElements(allSetupElements);
+        return allSetupElements;
+    }
+    
+    private void extractElements(ArrayList<BrowserSetupElement> allSetupElements) {
+       
+        for (BrowserSetupElement browserSetupElement : allSetupElements) {
+            List<BrowserSetting> capabilities = new ArrayList<BrowserSetting>();
+            List<BrowserSetting> options = new ArrayList<BrowserSetting>();
+            JsonObject jsonObject = browserSetupElement.getSettingsAsJsonObject();
+            setBrowserSetting(browserSetupElement, options, jsonObject, optionsString);  
```

Comment:
data flow is a bit obscure. capabilites is created here and filled elsewhere and assigned to the browserSetupElement elsewhere, too.